core/templates/pages/moderator-page/moderator-page.component.html

```diff
@@ -47,7 +47,7 @@ <h3>Recent Feedback Messages</h3>
             <tr ng-repeat="message in $ctrl.allFeedbackMessages track by $index">
               <td><[$ctrl.getDatetimeAsString(message.createdOnMsecs)]></td>
               <td>
-                <a ng-if="$ctrl.isEntityExploration(message.entityType)"
+                <a ng-if="$ctrl.isMessageFromExploration(message)"
```

Comment:
@srijanreddy98 We would have caught this with testing components via the HTML template, right(maybe)?

Reply:
I'm not too sure about this case because there is no user action associated to displaying anchor tags. I think using the DOM for testing would make more sense for actions e.g. clicking on something triggers a function.

Reply:
@DubeySandeep Yes, we would have caught this.

@kevintab95 while it is true that writing DOM tests for User Interactions makes more sense, it is just better to write tests for everything in the DOM. Taking the example of types here. In the ts files, we write `name: string = 'Some name';` The type string is implied from the value assigned to it. We don't need to explicitly imply it as typescript knows that the type is a string without us mentioning it. But we still do it so that contributors don't have to always think if the type is implied or not. Similarly, testing everything from the HTML is just better. Because everything in the ts file is written to aid some logic in the HTML file. So, all we have to do add two more lines to trigger the logic from the DOM.

Reply:
Not sure I understand exactly. Taking this particular issue as an example, do you mean we could have a test that asserts that the DOM has loaded / initialised a certain way with the data provided?

Reply:
Yes.